gms/gossiper.cc

```diff
@@ -1888,7 +1888,8 @@ future<> gossiper::do_shadow_round(std::unordered_set<gms::inet_address> nodes)
             gms::application_state::DC,
             gms::application_state::RACK,
             gms::application_state::SUPPORTED_FEATURES,
-            gms::application_state::SNITCH_NAME}};
+            gms::application_state::SNITCH_NAME,
+            gms::application_state::QUARANTINED_HOSTS}};
```

Comment:
More stuff added to gossiper - more stuff to remove from it in Raft. Can we perhaps only support quarantined hosts with Raft? @tzach Do we really need to solve this problem of returning zombie hosts in the old mode (no raft) given we're making Raft generally available?

Reply:
How soon is soon?
We already do partially successful job at fencing off zombie nodes that try to rejoin the cluster and have a couple of issues opened about that that I know of, plus we have dtests that test that, they would fail with just making the minimum change in replace not to inherit the host_id but with no other means of fencing the replaced node.

Reply:
Be it a week, a month or a year - it's still less than over a decade of Cassandra & Scylla life time this issue has been around. In any case I think that there are two parts of the pull request -one is a cleanup/bugfix and another is a feature. Given the cleanups are trivial, and there is so much in raft depending on them, it would be nice to get them in ASAP.